src/GafferSceneUI/ParameterInspector.cpp

```diff
@@ -167,6 +172,39 @@ Inspector::EditFunctionOrFailure ParameterInspector::editFunction( Gaffer::EditS
 {
 	auto attributeHistory = static_cast<const SceneAlgo::AttributeHistory *>( history );
 
+	/// \todo This duplicates the logic from `::parameterValue` in `EditScopeAlgo`. Should that
+	/// be exposed publicly in some way?
+
+	auto shaderNetwork = runTimeCast<const ShaderNetwork>( attributeHistory->attributeValue );
+	if( !shaderNetwork )
+	{
+		return fmt::format( "Attribute \"{}\" is not a shader.", attributeHistory->attributeName.string() );
+	}
+
+	const IECoreScene::Shader *shader;
+	if( m_parameter.shader.string().size() )
+	{
+		shader = shaderNetwork->getShader( m_parameter.shader );
+		if( !shader )
+		{
+			return fmt::format( "Shader \"{}\" does not exist.", m_parameter.shader.string() );
+		}
+	}
+	else
+	{
+		shader = shaderNetwork->outputShader();
+		if( !shader )
+		{
+			return "Output shader does not exist.";
+		}
+	}
+
+	const Data *result = shader->parametersData()->member( m_parameter.name );
+	if( !result )
+	{
+		return fmt::format( "Parameter \"{}\" does not exist.", m_parameter.name.string() );
+	}
```

Comment:
Rather than duplicate all this, I'd be very tempted to just call `ParameterInspector::value()` and use a generic "Parameter does not exist" message for null. Longer term I think we probably want to have some sort of registry of default values, so that we can create parameters from out of the blue, like we do for some attributes.